### PR TITLE
test(api): add types to conftest, test_simulate, test_execute

### PR DESCRIPTION
**Pull request by mcous**
## Overview

Happy tech debt Friday! More type annotation fixes

## Changelog

- test(api): add types to conftest, test_simulate, test_execute

## Review requests

Changes look good and are limited to tests

## Risk assessment

Very low

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
Nice, thank you!

File: api/tests/opentrons/conftest.py [hunk 1]
```diff
@@ -115,13 +154,14 @@ async def enable_ot3_hardware_controller(request):
...
-    except Exception:
-        root = request.param
+@pytest.fixture()
+def protocol_file() -> str:
+    return "testosaur_v2.py"
 
+
+@pytest.fixture()
+def protocol(protocol_file: str) -> Generator[Protocol, None, None]:
+    root = protocol_file
```
Comment (SyntaxColoring) [inline api/tests/opentrons/conftest.py:164]: This is a use of fixtures that I haven't seen before, so I might just need to learn how this works, but is this safe with the way `test_simulate_aspirate_tip()` "customizes" `protocol_file`? It looks like the same number of tests are running with these changes, but it's unclear to me if `test_simulate_aspirate_tip()`'s customization of `protocol_file` is propagating to also customize `protocol`.

**Review by SyntaxColoring — APPROVED** [review]
LGTM assuming CI passes! ⌨️  🏁

File: api/tests/opentrons/conftest.py (hunk 1 above)
Comment (mcous) [reply api/tests/opentrons/conftest.py:164]: Yup, [you can do this](https://docs.pytest.org/en/6.2.x/fixture.html#override-a-fixture-with-direct-test-parametrization)! It is cool / scary. I find this more readable than the previous version, but I'm happy to revert or do something else. Regardless, I don't think this fixture should be all the way up here in this `conftest.py`, but I didn't want to get _too_ crazy in this PR

**Pull request merged**